src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -332,6 +342,41 @@ private static bool IsAssemblyPoisoned(string path)
             return false;
         }
 
+        private static bool IsAssemblyFromSbrp(string assemblyPath)
+        {
+            using var stream = new FileStream(assemblyPath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+            using var peReader = new PEReader(stream);
+
+            MetadataReader reader = peReader.GetMetadataReader();
+            return reader.CustomAttributes.Select(attrHandle => reader.GetCustomAttribute(attrHandle))
+                    .Any(attr => IsAttributeSbrp(reader, attr));
+        }
+
+        private static bool IsAttributeSbrp(MetadataReader reader, CustomAttribute attr)
+        {
+            string attributeType = string.Empty;
+
+            if (attr.Constructor.Kind == HandleKind.MemberReference)
+            {
+                MemberReference mref = reader.GetMemberReference((MemberReferenceHandle)attr.Constructor);
+
+                if (mref.Parent.Kind == HandleKind.TypeReference)
+                {
+                    TypeReference tref = reader.GetTypeReference((TypeReferenceHandle)mref.Parent);
+                    attributeType = $"{reader.GetString(tref.Namespace)}.{reader.GetString(tref.Name)}";
+                }
+            }
+
+            if (attributeType == SbrpAttributeType)
+            {
+                byte[] data = reader.GetBlobBytes(attr.Value);
+                string attributeValue = Encoding.UTF8.GetString(data);
+
+                return attributeValue.Contains(SbrpAttributeValue);
```

Comment:
I realized that sometimes the string would be "sourcesource-build-reference-packages" and sometimes it would be "source source-build-reference-packages". Checking if the string contained "source-build-reference-packages" felt like the best solution to this problem.

Reply:
`source` is the name of the metadata key. Is there a way to get just the metadata value, which would allow you to compare for equality?

Reply:
> Is there a way to get just the metadata value, which would allow you to compare for equality?

Not to my knowledge.

With the given metadata reader logic and CustomAttribute logic, `attr.Value` returns the entire attribute value in a singular string, regardless of the attribute type. As such, in order to separate out the metadata key from the metadata value, the code would look something like this:

```
...
string attributeValue = Encoding.UTF8.GetString(data);
string result = attributeValue.Replace("source, "").Replace(" ", "");
return result == SbrpAttributeValue
```

IMO, the above code is similar to using `.Contains(SbrpAttributre)` but with extra logic.

Reply:
It's strange that it doesn't represent the value consistently. Perhaps using a regular expression against the whole value would be better. That way you can easily account for the optional space.

e.g. `source\s?source-build-reference-packages`

Reply:
You are sure that the assembly metadata is correct? I would be very surprised if this API does return assembly metadata deterministically.

Reply:
Yes, I'm sure. The attribute values contain non-readable characters, resulting in the need to parse out these characters and use a regex for the attribute match like Matt suggested. This change was made in this commit: https://github.com/dotnet/installer/pull/17339/commits/764bff31b89fc535789d03aff8a8b55099a334c3

Reply:
@ellahathaway please follow up (open an issue) against this API regarding the non-deterministic results.  This is hard to fathom why this would be the intended behavior.